src/diffusers/pipelines/ddim/pipeline_ddim.py

```diff
@@ -82,14 +82,15 @@ def __call__(
         # set step values
         self.scheduler.set_timesteps(num_inference_steps)
 
-        for t in self.progress_bar(self.scheduler.timesteps):
+        for step in self.progress_bar(self.scheduler.timesteps):
             # 1. predict noise model_output
+            t = self.scheduler.get_noise_condition(step)
```

Comment:
it's confusing to me that `get_noise_condition( )` return a `time` integer. Also why is this need for DDIM?